### PR TITLE
Implement maximum text length constraint for interval input field

### LINKED ISSUE #3349: Unlimited repeats in Planned payments Screen hide "Time Component"
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

In **planned payments** repeat section, there are unlimited repeats for (day, week, month or year) and as we keeping entering numbers, "repeats every" box keeps on stretching across screen which ultimately pushes time (day, week, month or year) button off screen.

### To Reproduce

1. Go to planned payments screen
2. Select multiple times
3. Keep on adding "repeats every" number

### Expected behavior

In my opinion, there should be multi-line feature for large numbers so that time button stays visible OR there should be a limit of repeats corresponding to time (Day, Week, Month or Year).

### Screenshots

Screenshot_20240717-181045_Ivy Wallet
Screenshot_20240717-181056_Ivy Wallet

### App version

4.6.3 (163) 

### Smartphone

Samsung Galaxy A12, Android 12

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by akashs056**
## Pull request (PR) checklist
Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [x] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?
- Set a maximum text length of three digits for the interval input field.
- Adjusted the button width to ensure it stays perfectly within the screen.
- Applied the logic that the maximum interval can only be up to 365 days, which fits within three digits.

Before

https://github.com/user-attachments/assets/25168f98-4892-4df5-ac43-5d6673082618

After

https://github.com/user-attachments/assets/5dcb4e8a-0262-489d-8d46-4ba597e360db

## Does this PR close any GitHub issues? (do not delete)
- Closes #3349

### CODE REVIEW

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt [hunk 1]
```diff
@@ -58,6 +58,7 @@ fun IntervalPickerRow(
 
         IvyNumberTextField(
             modifier = Modifier
+                .width(125.dp)
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt:61]: Remove this - it won't scale well for users using the largest font on Android

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt [hunk 2]
```diff
@@ -73,13 +74,14 @@ fun IntervalPickerRow(
             textColor = if (validInput) White else UI.colors.pureInverse,
             hint = "0"
         ) {
+            val filteredText = if (it.text.length <= 3) it.text else it.text.take(3)
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt:77]: Extract `3` as a constant (Detekt told you probably) and `it.text.take(3)` is sufficient  - no need for an `if`.

Additionally change the constant to 5 to give more room for custom stuff

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt (hunk 2 above)
Comment (akashs056) [reply temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt:77]: @ILIYANGERMANOV In my opinion if we are limiting the entered amount then its no need for that width attribute or any other changes to the UI we can just it.text.take(3) make this 5, as you suggested . Additionally i have a question about where should i declare the constant . There are multiple constants file in the application or should i declare in the same file itself ? 
image

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt:77]: @akashs056 in the same file where you make the change. At the top of the file

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/IntervalPickerRow.kt:77]: it.text.take(RepeatIntervalCharLimit)

**Review by ILIYANGERMANOV — APPROVED** [review]

**Comment by akashs056** [comment]
@ILIYANGERMANOV Its ready to merge

**Review by MudassarStd — APPROVED** [review]

**Pull request merged**